javascript/javascript1/week2/formalFlight.js

```diff
@@ -0,0 +1,20 @@
+//Flight booking fullname function(Formal fullname)
+let firstName= prompt("What is your first name?");
+let lastName= prompt("What is your last name?");
+let useFormalName =prompt("Do you whant to adress formal? (yes or no)")
+let gender= prompt("What is your gender? (male or female)")
+
+function getFullname(firstName,lastName,useFormalName,gender){
+    if (useFormalName.toLowerCase() === "yes" && gender.toLowerCase()==="male"){
+        return "Lord "  + firstName + " "+ lastName;
+    }
+
+    else if(useFormalName.toLowerCase() === "yes" && gender.toLowerCase()==="female"){
+        return "Madam " + firstName + " "+ lastName;
+    }
+
+    else{ 
+        return firstName + " "+ lastName;}
+}
+
+console.log(getFullname(firstName,lastName,useFormalName,gender));
```

Comment:
Since the first two conditions take useFormalName as "yes", you can put it in one condition and nest the rest inside.
And if you want it to be a bit fancier, you wouldn't need to ask for the gender if the answer is no! below you can find a suggestion attached to how it could also be! 

```suggestion
let gender;

function getFullname(firstName, lastName, useFormalName) {
  if (useFormalName.toLowerCase() === "yes") {
    gender = prompt("What is your gender? (male or female)");
    if (gender.toLowerCase() === "male" || gender.toLowerCase() === "m") {
      return "Mr. " + firstName + " " + lastName;
    } else if (
      gender.toLowerCase() === "female" ||
      gender.toLowerCase() === "f"
    ) {
      return "Mrs. " + firstName + " " + lastName;
    }
  } else {
    return firstName + " " + lastName;
  }
}

console.log(getFullname(firstName, lastName, useFormalName));

```

Reply:
Thanks. Your recommendation is really professional and I learnt good points :)